llvm/include/llvm/ADT/APInt.h

```diff
@@ -113,13 +116,21 @@ class [[nodiscard]] APInt {
       : BitWidth(numBits) {
     if (!implicitTrunc) {
       if (BitWidth == 0) {
-        assert(val == 0 && "Value must be zero for 0-bit APInt");
+        // EraVM local begin
+        if (val != 0)
+          report_fatal_error("APInt error: Value must be zero for 0-bit APInt");
+        // EraVM local end
       } else if (isSigned) {
-        assert(llvm::isIntN(BitWidth, val) &&
-               "Value is not an N-bit signed value");
+        // EraVM local begin
+        if (!llvm::isIntN(BitWidth, val))
+          report_fatal_error("APInt error: Value is not an N-bit signed value");
+        // EraVM local end
       } else {
-        assert(llvm::isUIntN(BitWidth, val) &&
-               "Value is not an N-bit unsigned value");
+        // EraVM local begin
+        if (!llvm::isUIntN(BitWidth, val))
```

Comment:
I'm not sure this will catch bitwidth larger than 64, since `isUIntN` and `isIntN` will return true for BitWidth >= 64. Please take a look [here](https://github.com/matter-labs/era-compiler-llvm/blob/0482c7a514b7e9dfec713edf5d1ea204249dbcd4/llvm/include/llvm/Support/MathExtras.h#L228).

Reply:
FWIW, I think we should make this change, since it can catch some miscompilation issues.

Reply:
What change do you refer to?

Reply:
To change asserts with fatal errors, not the last change you introduced as experiment AFAIU.